plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -449,22 +455,19 @@ public Optional<JoinApplicationResult<ConnectorTableHandle>> applyJoin(
         if (!leftHandle.getAuthorization().equals(rightHandle.getAuthorization())) {
             return Optional.empty();
         }
+
         int nextSyntheticColumnId = max(leftHandle.getNextSyntheticColumnId(), rightHandle.getNextSyntheticColumnId());
 
         ImmutableMap.Builder<JdbcColumnHandle, JdbcColumnHandle> newLeftColumnsBuilder = ImmutableMap.builder();
         for (JdbcColumnHandle column : jdbcClient.getColumns(session, leftHandle)) {
-            newLeftColumnsBuilder.put(column, JdbcColumnHandle.builderFrom(column)
-                    .setColumnName(column.getColumnName() + "_" + nextSyntheticColumnId)
-                    .build());
+            newLeftColumnsBuilder.put(column, aliasFormatter.format(column, nextSyntheticColumnId, session));
             nextSyntheticColumnId++;
         }
         Map<JdbcColumnHandle, JdbcColumnHandle> newLeftColumns = newLeftColumnsBuilder.buildOrThrow();
 
         ImmutableMap.Builder<JdbcColumnHandle, JdbcColumnHandle> newRightColumnsBuilder = ImmutableMap.builder();
         for (JdbcColumnHandle column : jdbcClient.getColumns(session, rightHandle)) {
-            newRightColumnsBuilder.put(column, JdbcColumnHandle.builderFrom(column)
-                    .setColumnName(column.getColumnName() + "_" + nextSyntheticColumnId)
-                    .build());
+            newRightColumnsBuilder.put(column, aliasFormatter.format(column, nextSyntheticColumnId, session));
```

Comment:
here you should pass maximum identifier length as OptionalInt acquired from jdbcClient method with the default implementation:

`newRightColumnsBuilder.put(column, aliasFormatter.format(session, column, jdbcClient.getMaximumColumnIdentifierLength(), nextSyntheticColumnId));`